### DISCUSSION TITLE
Custom Sorting strategy for igx grid

### DISCUSSION THREAD

**Discussion by Vighneshaa** [IgniteUI/igniteui-angular#13457] [Q&A] [answered]
Where can I find the documentation for the custom Sorting strategy in igx?
Also, Can you provide me with the example related to custom Sorting strategy?

**Comment by ddaribo** [comment, marked as answer]
Hello @Vighneshaa,

The ["Sorting through the API" section](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sorting#sorting-through-the-api) in the IgxGrid documentation outlines a few key aspects about providing a custom implementation of the `ISortingStrategy` interface.

As you can read there, "_Any [IgxColumnComponent](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxcolumncomponent.html#sortStrategy) or [ISortingExpression](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/isortingexpression.html#strategy) can use a custom implementation of the [ISortingStrategy](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/isortingstrategy.html) as a substitute algorithm_".

[Here](https://stackblitz.com/edit/igx-grid-custom-sort-strategy?file=src%2Fapp%2Fgrid%2Fgrid-sorting-sample%2Fgrid-sorting-sample.component.ts) is a StackBlitz sample demonstrating a simple custom sort strategy applied on two columns of the IgxGrid (the initial sorting expression on the `CategoryName` column as well as the `sortStrategy` assigned to the `ShipCountry` column). For the purposes of the example, the sorting algorithm is altered to sort by the last symbol, in case the compared values are strings and are non-empty. Of course, you could implement any comparison logic that would be relevant to your requirement.
For reference, the implementation of the `DefaultSortingStrategy` can be found in the source code [here](https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/data-operations/sorting-strategy.ts#L31).
